core/src/main/kotlin/live/LiveGuild.kt

```diff
@@ -19,6 +20,124 @@ import dev.kord.core.event.user.VoiceStateUpdateEvent
 @KordPreview
 fun Guild.live(): LiveGuild = LiveGuild(this)
 
+@KordPreview
+inline fun Guild.live(block: LiveGuild.() -> Unit) = this.live().apply(block)
+
+@KordPreview
+fun LiveGuild.onEmojisUpdate(block: suspend (EmojisUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onIntegrationsUpdate(block: suspend (IntegrationsUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onBanRemove(block: suspend (BanRemoveEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onPresenceUpdate(block: suspend (PresenceUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onVoiceServerUpdate(block: suspend (VoiceServerUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onVoiceStateUpdate(block: suspend (VoiceStateUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onWebhookUpdate(block: suspend (WebhookUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onRoleCreate(block: suspend (RoleCreateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onRoleUpdate(block: suspend (RoleUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onRoleDelete(block: suspend (RoleDeleteEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberJoin(block: suspend (MemberJoinEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberUpdate(block: suspend (MemberUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberLeave(block: suspend (MemberLeaveEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+inline fun LiveGuild.onReaction(crossinline block: suspend (Event) -> Unit) = on<Event> {
+    if (it is ReactionAddEvent || it is ReactionRemoveEvent) {
+        block(it)
+    }
+}
+
+@KordPreview
+inline fun LiveGuild.onReaction(
+    emoji: ReactionEmoji,
+    crossinline block: suspend (Event) -> Unit
+) = on<Event> {
+    if (it is ReactionAddEvent && (emoji == it.emoji) || it is ReactionRemoveEvent && (emoji == it.emoji)) {
+        block(it)
+    }
+}
```

Comment:
This method allows to make a behavior when there is an event related to the reaction regardless of this event. I think it might be useful for some developer to do: "If there is an interaction (add or remove) with this reaction, then I do this".

This is to avoid duplicating code for example in : 
```kotlin
onReactionAdd { callFunction() }
onReactionRemove {callFunction() }
```

But just do that :

```kotlin
onReaction { callFunction() }
```

Reply:
I was looking for a more concrete use case. The way I see it, you can draw contextual relations between any two events.
Not sure this is something we should cater to when the alternative implementation consists of one more line.

Reply:
The concrete case will be created by the user. Currently, I do not need it. However I think that for some people depending on the context it can be useful.
But if you think it's better to take it off, I can understand

Reply:
I think these sort of functions are more of a workaround for problems with the events hierarchy and they're better solved there. You want an event listener that triggers on events related to a (specific) reaction being used, that should be represented on the event level and type hierarchy instead, right now you're losing all information on what the event actually is. There's also no reason these sort of event listeners shouldn't also exist on the Kord class itself, any use-case motivated here could be motivated there as well.

So yes, I'd rather not see them implemented here, it's a partial solution to a deeper problem.

Reply:
Yes i understand. So the methods onReaction on each LiveEntity has been removed